pmlc/dialect/tile/builder.h

```diff
@@ -44,6 +45,21 @@ struct ProgramUpdate {
 struct ProgramMutations {
   std::vector<mlir::Value> outputs;
   std::set<ProgramUpdate> updates;
+  DataType floatx = DataType::invalid;
+  DataType intx = DataType::invalid;
+
+  void set_floatx(const char* floatx_str) {
+    floatx = util::from_string(floatx_str);
+    if (!isFloat(floatx)) {
+      throw std::runtime_error("Invalid floatx string " + std::string(floatx_str));
+    }
+  }
+  void set_intx(const char* intx_str) {
+    intx = util::from_string(intx_str);
+    if (!isInteger(intx)) {
+      throw std::runtime_error("Invalid intx string " + std::string(intx_str));
+    }
+  }
```

Comment:
I think we should pass the `floatx` and `intx` as parameters to the `MakeProgram()` method instead of attaching them to the `ProgramMutations` (which should only concern itself with specifying the outputs and updates aka mutations of the program)